llvm/lib/Transforms/IPO/DeadArgumentElimination.cpp

```diff
@@ -1268,5 +1269,70 @@ void DeadArgumentEliminationPass::UpdateNVPTXMetadata(Module &M, Function *F,
       continue;
     // Update the metadata with the new function
     MetadataNode->replaceOperandWith(0, llvm::ConstantAsMetadata::get(NF));
+
+    // Carefully update any and all grid_constant annotations, since those are
+    // denoted parameter indices, which may have changed
+    for (unsigned i = 1; i < MetadataNode->getNumOperands() - 1; i += 2) {
```

Comment:
```suggestion
    for (unsigned I = 1; I < MetadataNode->getNumOperands() - 1; I += 2) {
```

Reply:
Ah thanks, wasn't sure if we went with that style or not.